scripts/lesson_run.py

```
@@ -0,0 +1,233 @@
+def syntax_check(py_files: list[Path]) -> tuple[bool, str]:
+    for path in py_files:
+        try:
+            py_compile.compile(str(path), doraise=True)
+        except py_compile.PyCompileError as exc:
+            return False, f"{path.relative_to(ROOT).as_posix()}: {exc.msg.strip()}"
+    return True, ""
+
+
+def execute_lesson(entry: Path) -> tuple[bool, str]:
+    try:
+        proc = subprocess.run(
```

Ruff (0.15.13)

[error] `subprocess` call: check for execution of untrusted input

(S603)